### ISSUE TITLE
Watchmarker on right side (recommended?) no longer showing, even after manual sync

### ISSUE THREAD

**Issue by Mycohl** [sniklaus/youtube-watchmarker#134]
Appears to be happening in Firefox and Chrome.

This may be a duplicate of #114 but I'm not sure.

I've performed a sync by clicking the button under the "Youtube" section, but the issue persists.

![watch-marker](https://github.com/user-attachments/assets/bd913623-16a4-4b2a-a08e-4927ed25e072)

**Comment by sniklaus** [comment]
I just released version `4.7.0` (already live on Firefox, still pending on Chrome), did this update resolve your issue by chance?

**Comment by Mycohl** [comment]
> did this update resolve your issue by chance?

It's difficult to determine, without knowing the nature of the issue or what was changed.

I still have frequent absence of the Watchmarker, although now I find that manually syncing resolves the issue. However, I don't feel that I should have to manually sync my browser history every 48 hours. Why is the extension failing to read my browser history? (This is on a Firefox fork, for the record.)

Additionally, I'm now seeing the Youtube "watched" bar on top of the Watchmarker. This wasn't the case before. If this is intended behavior, I can probably deal with it, but having to manually sync every couple of days is tiresome.

![Image](https://github.com/user-attachments/assets/aeefcf06-8d37-4d6a-9d7b-80e2d72aea65)

**Comment by sniklaus** [comment]
That are a bunch of weird behaviors that I have neither experienced myself nor do I recall having others reported them. 🤔 

What is the Firefox fork that you are using? Are they doing some privacy enhancement that might interfere with the background scripts of extensions?

**Comment by Mycohl** [comment]
> That are a bunch of weird behaviors

Another thing I just noticed is that the Watchmarker is appearing over videos that I've only opened in private/incognito windows. These shouldn't be appearing in my browser history, correct? And Youtube shouldn't be reporting them as "watched" by my account, correct? This seems like a bug.

> What is the Firefox fork that you are using? 

I'm using both Waterfox and Floorp.

> Are they doing some privacy enhancement that might interfere with the background scripts of extensions?

I honestly couldn't tell you. If there are particular logs or other information you would like me to collect, I'm happy to do so.

**Comment by sniklaus** [comment]
> Another thing I just noticed is that the Watchmarker is appearing over videos that I've only opened in private/incognito windows. These shouldn't be appearing in my browser history, correct? And Youtube shouldn't be reporting them as "watched" by my account, correct? This seems like a bug.

Do you have "mark a video as watched once it has been opened in the browser" enabled in the settings? If so then it would also be added to the Watchmarker history in private sessions. So either disable that setting or don't allow the extension to run in private windows. 👍 

> I'm using both Waterfox and Floorp. I honestly couldn't tell you. If there are particular logs or other information you would like me to collect, I'm happy to do so. If you'd like me to attempt to repro with Ubuntu's repo firefox, I can work on getting that set up.

I do wonder whether vanilla Firefox would behave as expected. 🤔

**Comment by Mycohl** [comment]
>  So either disable that setting or don't allow the extension to run in private windows. 👍

I'm not sure how it was enabled for private windows to begin with, but I've disabled that now.
 
Had to toggle it off and on again, I do notice that it happens after I reboot, not just when I restart the browser. I will try other tests.

> I do wonder whether vanilla Firefox would behave as expected. 🤔

I will work on getting that set up sometime this week, hopefully.

**Comment by sniklaus** [comment]
Might just be some conflict between the mainline version and the fork, there are some discussions at #171 on how to resolve it. 👍
